### ISSUE TITLE
NextJS empty POST request object and "Failed to parse JSON"

### ISSUE THREAD

**Issue by peacefulotter** [oven-sh/bun#7336]
### What version of Bun is running?

1.0.14+1840de366

### What platform is your computer?

Linux 6.2.0-37-generic x86_64 x86_64

### What steps can reproduce the bug?

```sh
$ bun create next-app
$ cd my-app
$ bun --bun run dev
```
In `page/app.tsx`:
```tsx
'use client'
import { useEffect } from 'react'

export default function Home() {
    useEffect(() => {
        // Sending a POST request with a JSON body
        fetch('/api/', {
            method: 'POST',
            headers: {
                'Content-Type': 'application/json',
            },
            body: JSON.stringify({
                foo: 'bar' 
            }),
        })
            .then((response) => response.json())
            .then((data) => console.log(data))
    }, [])
    return ...
}
```
In  `page/api/route.ts`:
```ts
export async function POST(req: Request) {
    console.log(req)
    const res = await req.json()
    console.log(res)
    return Response.json({ ok: 'ok' })
}
```

### What is the expected behavior?

`req` is not an empty object and `await req.json()` returns a json object without throwing an error.

### What do you see instead?

```js
{} // From the console.log(req)


 ⨯ 1 | /* harmony export */ __webpack_require__.d(__webpack_exports__, {
2 | /* harmony export */   POST: () => (/* binding */ POST)
3 | /* harmony export */ });
4 | async function POST(req) {
5 | 
6 |     const res = await req.json();
                         ^
SyntaxError: Failed to parse JSON
      at POST (:5:20)
      at /home/edge/dev/js/bun-nextjs-test/node_modules/next/dist/compiled/next-server/app-route.runtime.dev.js:6:62613
      at /home/edge/dev/js/bun-nextjs-test/node_modules/next/dist/compiled/next-server/app-route.runtime.dev.js:6:56279
      at /home/edge/dev/js/bun-nextjs-test/node_modules/next/dist/server/lib/trace/tracer.js:131:35
      at run (node:async_hooks:70:10)
      at wrap (/home/edge/dev/js/bun-nextjs-test/node_modules/next/dist/compiled/next-server/app-route.runtime.dev.js:6:36757)
      at run (node:async_hooks:70:10)
      at wrap (/home/edge/dev/js/bun-nextjs-test/node_modules/next/dist/compiled/next-server/app-route.runtime.dev.js:6:36285)
      at run (node:async_hooks:70:10)
      at /home/edge/dev/js/bun-nextjs-test/node_modules/next/dist/compiled/next-server/app-route.runtime.dev.js:6:53878
      at execute (/home/edge/dev/js/bun-nextjs-test/node_modules/next/dist/compiled/next-server/app-route.runtime.dev.js:6:53658)
      at /home/edge/dev/js/bun-nextjs-test/node_modules/next/dist/compiled/next-server/app-route.runtime.dev.js:6:63851
      at handle (/home/edge/dev/js/bun-nextjs-test/node_modules/next/dist/compiled/next-server/app-route.runtime.dev.js:6:63834)
      at /home/edge/dev/js/bun-nextjs-test/node_modules/next/dist/server/base-server.js:1331:47
      at doRender (/home/edge/dev/js/bun-nextjs-test/node_modules/next/dist/server/base-server.js:1256:32)
      at /home/edge/dev/js/bun-nextjs-test/node_modules/next/dist/server/base-server.js:1567:33
      at /home/edge/dev/js/bun-nextjs-test/node_modules/next/dist/server/base-server.js:1475:76
      at get (/home/edge/dev/js/bun-nextjs-test/node_modules/next/dist/server/response-cache/index.js:46:14)
      at /home/edge/dev/js/bun-nextjs-test/node_modules/next/dist/server/base-server.js:1475:33
      at renderToResponseWithComponentsImpl (/home/edge/dev/js/bun-nextjs-test/node_modules/next/dist/server/base-server.js:999:47)
      at /home/edge/dev/js/bun-nextjs-test/node_modules/next/dist/server/base-server.js:983:192
      at renderToResponseWithComponents (/home/edge/dev/js/bun-nextjs-test/node_modules/next/dist/server/base-server.js:982:41)
      at /home/edge/dev/js/bun-nextjs-test/node_modules/next/dist/server/base-server.js:1852:29
      at processTicksAndRejections (:61:76)
```

### Additional information

Running the app with `bun run dev` (without --bun) works fine.

package.json (untouched after `bun create next-app`):
```json
"dependencies": {
    "react": "^18",
    "react-dom": "^18",
    "next": "14.0.3"
  },
  "devDependencies": {
    "typescript": "^5",
    "@types/node": "^20",
    "@types/react": "^18",
    "@types/react-dom": "^18",
    "autoprefixer": "^10.0.1",
    "postcss": "^8",
    "tailwindcss": "^3.3.0",
    "eslint": "^8",
    "eslint-config-next": "14.0.3"
  }
```

**Comment by Electroid** [comment]
@Jarred-Sumner This could to be the same bug that we found with robobun, where the request body is cloned, then read. This assumes Next is cloning the body, which is entirely plausible.

**Comment by peacefulotter** [comment]
@Electroid Any workaround for now? Or plans to fix this soon?

**Comment by akoskovacs** [comment]
This also seems to be an issue when you try to send `FormData` objects to the application.

**Comment by camiloux** [comment]
I have the same issue with bun in Astro. Works fine with bun run dev, but not on the built application. I simply get an empty formdata and I haven't been able to find a workaround. It works fine when running on node though. Please fix this, I makes Bun not usable for production when having forms (with Astro and other frameworks).

**Comment by matheusburey** [comment]
I updated my bun to version 1.0.25 and it fixed my problem

**Comment by Electroid** [comment]
If you still run into this issue after upgrading Bun, let us know and we'll reopen the issue.

**Comment by DragonOfPou** [comment]
I still have the Problem with the latest Version of Next.js. I tried Bun 1.0.26 and 1.0.25. Im Running on 8.0.36-0ubuntu0.22.04.1 (Ubuntu) on Windows WSL. Without --bun or with npm i get no error.

**Comment by raomeiza** [comment]
I'm currently having thesame issue with nextjs v14 on nodejs v20.9.0 running in windows

**Comment by Jarred-Sumner** [comment]
Can you provide a more complete reproduction? Something I can clone and immediately see the error?

**Comment by peacefulotter** [comment]
This error seems to be fixed on v1.0.30.
It didn't work for me when running with `--bun` on v1.0.25.
Thanks for the support!
